### PR TITLE
util: Add a range check for HSVLike

### LINKED ISSUE #496: Add range checks to HSVLike?
https://github.com/KyoriPowered/adventure/blob/259578b0b13c38a532d09443e55360a08e55ea4e/api/src/main/java/net/kyori/adventure/util/HSVLike.java#L35

### PR DESCRIPTION
**Pull request by KingOfSquares**
Closes #496

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/util/Range.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A utility that can check if arbitrary numbers are inside a specified range.
+ *
+ * @since 4.10.0
+ */
+public interface Range {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/util/Range.java:34]: Not really a fan of exposing this at this time - would prefer we just inline the checks into a `private static` method in `HSVLikeImpl` for now.

**Review by kashike — CHANGES_REQUESTED** [review]

File: api/src/main/java/net/kyori/adventure/util/Range.java (hunk 1 above)
Comment (KingOfSquares) [reply api/src/main/java/net/kyori/adventure/util/Range.java:34]: Ok

**Review by zml2008 — APPROVED** [review]

**Review by kashike — APPROVED** [review]

File: api/src/main/java/net/kyori/adventure/util/HSVLikeImpl.java [hunk 2]
```diff
@@ -53,6 +56,13 @@ public float v() {
     return this.v;
   }
 
+  private static void requireInsideRange(final float number, final String name) throws IllegalArgumentException {
+    if (!(0 <= number && number <= 1)) {
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/util/HSVLikeImpl.java:60]: Why not or this?

File: api/src/main/java/net/kyori/adventure/util/HSVLikeImpl.java [hunk 3]
```diff
@@ -53,6 +56,13 @@ public float v() {
...
+      throw new IllegalArgumentException(
+        name + " (" + number + ")" + " is not inside the required range: [" + 0 + "," + 1 + "]");
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/util/HSVLikeImpl.java:62]: Inline these numbers?

**Review by kezz — CHANGES_REQUESTED** [review]

**Review by kezz — APPROVED** [review]

**Pull request merged**